### ISSUE TITLE
Join breaks PETL transformations on following objects (+ inconsistency on how it breaks...)

### ISSUE THREAD

**Issue by TinMarkovic** [petl-developers/petl#551]
#### Minimal, reproducible code sample, a copy-pastable example if possible

```python
import petl

table1 = [['foo', 'bar'],
          ['a', 1],
          ['a', 2],
          ['b', 3],
          ['b', 4],
          ['b', 5],
          ['c', 6]]

table2 = [['bar', 'baz'],
          [1, True],
          [2, False],
          [3, True],
          [4, False],
          [5, False],
          [6, True]]

table3 = petl.leftjoin(table1, table2, key="bar").cutout("bar")  # works
table3.cutout("baz")  # does not work
petl.cutout(table3, "baz")  # does not work

print(table3)
```

#### Problem description

I would expect to be able to continue using PETL transformations after joining two tables. In the current iteration, that is only possible if I do not assign the variable (this was quite confusing, and I discovered it while making the minimal reproduction). 

Other operations, like renaming and cutting, also do not work. I've tried using ``rightjoin`` as well, didn't change anything.

Thanks for looking into it!

#### Version and installation information

* ``petl.__version__ == 1.7.4``
* ``Python 3.8.5 (default, Jan 27 2021, 15:41:15) [GCC 9.3.0] on linux``
* Installed via pip into virtual environment

**Comment by dnicolodi** [comment]
I cannot reproduce this with Python 3.9 and petl 1.7.6. Also I don't see how this could be possible, at the Python interpreter level, `a = foo().bar()` and `t = foo(); a = t.bar()` are the exactly the same thing (except that the second adds a `t` variable to the local scope).

Has anyone confirmed the bug?

**Comment by arturponinski** [comment]
Created a test case for this, cannot reproduce:
```python
import petl

    table1 = [['foo', 'bar'],
              ['a', 1],
              ['a', 2],
              ['b', 3],
              ['b', 4],
              ['b', 5],
              ['c', 6]]

    table2 = [['bar', 'baz'],
              [1, True],
              [2, False],
              [3, True],
              [4, False],
              [5, False],
              [6, True]]


    table3 = petl.leftjoin(table1, table2, key="bar").cutout("bar")  # works
    expect = [['foo'],
              ['a'],
              ['a'],
              ['b'],
              ['b'],
              ['b'],
              ['c']]
    table4 = table3.cutout("baz")  # does not work
    ieq(expect, table4)
    table5 = petl.cutout(table3, "baz")  # does not work
    ieq(expect, table5)
```
The provided code does work as expected.
In case we've missed something please reopen and provide a failing test-case (a PR with failing test would be appreciated most)

**Comment by TinMarkovic** [comment]
So, I've checked it again, and it works as expected. I am quite confused, because as another comment explained:

> Also I don't see how this could be possible, at the Python interpreter level

I am not curious enough to debug it by going down and recreating the full environment, and am happy that the quirk disappeared :sweat_smile: 

I've discovered this during a very short experiment using petl, sorry for not communicating earlier/clearer. Thanks everyone for your attention.
